WalletWasabi/WabiSabi/Backend/Statistics/TxPropagationVerifier.cs

```diff
@@ -0,0 +1,65 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin;
+using WalletWasabi.Logging;
+using WalletWasabi.WabiSabi.Backend.WebClients;
+
+namespace WalletWasabi.WabiSabi.Backend.Statistics;
+
+public class TxPropagationVerifier
+{
+	public TxPropagationVerifier(Network network, IHttpClientFactory httpClientFactory)
+	{
+		Network = network;
+		Verifiers = new List<ITxPropagationVerifier>()
+		{
+			new BlockstreamApiClient(Network, httpClientFactory.CreateClient(nameof(BlockstreamApiClient))),
+			new MempoolSpaceApiClient(Network, httpClientFactory.CreateClient(nameof(MempoolSpaceApiClient)))
+		};
+	}
+	private Network Network { get; }
+	public List<ITxPropagationVerifier> Verifiers { get; }
+
+	// Don't throw as this task is not awaited
+	public async Task LogTxAcceptedByThirdPartyAsync(uint256 txid, CancellationToken cancel, int delay = 30000, int attempts = 10)
+	{
+		for (int i = 0; i < attempts; i++)
+		{
+			try
+			{
+				await Task.Delay(delay, cancel).ConfigureAwait(false);
```

Comment:
There are so many `Task.Delay`s in this PR, why do we need this one, too? 
There are delays inside the `ApiClients`.